sdk/python/foundation-models/system/inference/zero-shot-image-classification/zero-shot-image-classification-batch-endpoint.ipynb

```diff
@@ -0,0 +1,511 @@
+   "metadata": {},
+   "source": [
+    "#### 3.1 Arrange images in common folder for batch inference input"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dir_names = []\n",
+    "for dir_name in os.listdir(dataset_dir):\n",
+    "    dir_names.append(dir_name)\n",
+    "    dir_path = os.path.join(dataset_dir, dir_name)\n",
+    "    for path, subdirs, files in os.walk(dir_path):\n",
+    "        for file in files:\n",
+    "            shutil.move(os.path.join(path, file), dataset_dir)\n",
+    "\n",
+    "    shutil.rmtree(dir_path)\n",
+    "\n",
+    "labels = \", \".join(dir_names)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 3.2 Prepare a folder of CSV files with base64 images for batch inference input\n",
+    "\n",
+    "We can provide input images to batch inference in a csv file containing a column named \"image\" having images in base64 format. We provide the candidate labels in the same csv file in the first row of a column named \"text\".\n",
+    "The deployment in the `Create batch deployment` section below takes the argument `mini_batch_size`, which is the number of CSV files processed by the model in a single mini_batch. To limit the number of images processed in each mini_batch we split the dataset into multiple csv files."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import base64\n",
+    "import pandas as pd\n",
+    "\n",
+    "csv_folder_path = os.path.join(dataset_parent_dir, \"batch\")\n",
+    "os.makedirs(csv_folder_path, exist_ok=True)\n",
+    "batch_input_file = \"batch_input.csv\"\n",
+    "\n",
+    "image_list = []\n",
+    "for image in os.listdir(dataset_dir):\n",
+    "    with open(os.path.join(dataset_dir, image), \"rb\") as f:\n",
+    "        data = f.read()\n",
+    "        data = base64.encodebytes(data).decode(\"utf-8\")\n",
+    "        image_list.append(data)\n",
+    "\n",
+    "data = [[image, \"\"] for image in image_list]\n",
+    "batch_df = pd.DataFrame(data, columns=[\"image\", \"text\"])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Divide this into files of 10 rows each\n",
+    "batch_size_per_predict = 10\n",
+    "for i in range(0, len(batch_df), batch_size_per_predict):\n",
+    "    j = i + batch_size_per_predict\n",
+    "    batch_df[\"text\"].iloc[\n",
+    "        i\n",
+    "    ] = labels  # each csv file includes the labels in the first row\n",
+    "    batch_df[i:j].to_csv(os.path.join(csv_folder_path, str(i) + batch_input_file))\n",
+    "\n",
+    "# Check out the first and last file name created\n",
+    "input_files = os.listdir(csv_folder_path)\n",
+    "print(f\"{input_files[0]} to {str(i)}{batch_input_file}.\")"
```

Comment:
Nit: This print statement is printing the following:
"110batch_input.csv to 130batch_input.csv."
Should this be "0batch_input.csv to 130batch_input.csv."

Reply:
<img width="548" alt="image" src="https://github.com/Azure/azureml-examples/assets/38021489/7d9badd6-5d43-4e7e-b1ca-0d3b26ab6c41">
the print statement is accurate